sdk/login_form/login_form_handler.js

```diff
@@ -1,37 +1,240 @@
+  loginForm.addEventListener('submit', function (event) {
+    event.preventDefault();
+    const email = document.getElementById('web-nexus-email-input')
+      .value.trim();
+    const password = document.getElementById('web-nexus-password-input')
+      .value.trim();
+    // const salt = bcrypt.genSaltSync(10);
+    // const hashedPassword = await bcrypt.hash(password, salt);
+
+    login(email, password)
+      .then((data) => {
+        authEventHandler.onSuccessfulLoginHandler(data);
+        addUserOnline(email);
+      })
+      .catch((errorHandler) => {
+        errorHandler();
+      });
+  });
+}
+
+function setUpRegisterForm(divId) {
+  document.addEventListener("click", function (event) {
+    if (event.target && event.target.id === "switch-to-register") {
+      event.preventDefault();
+      fetch(registerFormUrl)
+        .then((result) => result.text())
+        .then((body) => {
+          const authContainerDiv = document.getElementById(divId);
+          authContainerDiv.innerHTML = body;
+          forms.registerForm = body;
+        })
+        .then(setUpRegisterEvent)
+        .then(setUpSwitchToLoginEvent(divId))
+        .catch((error) => {
+          console.log(error);
+        });
+    }
+  });
+}
+
+function setUpRegisterEvent() {
+  const registerForm = document.getElementById('web-nexus-register-form');
+  registerForm.addEventListener('submit', async function (event) {
+    event.preventDefault();
+    const firstName = document.getElementById('web-nexus-firstname-input')
+      .value.trim();
+    const lastName = document.getElementById('web-nexus-lastname-input')
+      .value.trim();
+    const email = document.getElementById('web-nexus-email-input')
+      .value.trim();
+    const password = document.getElementById('web-nexus-register-password-input')
+      .value.trim();
+    const salt = bcrypt.genSaltSync(10);
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+    register(firstName, lastName, email, hashedPassword)
+      .then((data) => {
+        authEventHandler.onSuccessfulLoginHandler(data);
+        addUserOnline(email);
+      })
+      .catch((errorHandler) => {
+        errorHandler();
+      });
+  });
+}
+
+function setUpSwitchToLoginEvent(divId) {
+  document.addEventListener("click", function (event) {
+    if (event.target && event.target.id === "switch-to-login") {
+      event.preventDefault();
+      const authContainerDiv = document.getElementById(divId);
+      authContainerDiv.innerHTML = forms.loginForm;
+    }
+  });
+}
+
+function setResponseEvents(onSuccessfulLoginCb, onErrorCb, onServerError) {
+  if (onSuccessfulLoginCb !== undefined) {
+    authEventHandler.onSuccessfulLoginHandler = onSuccessfulLoginCb;
+  }
+  if (onErrorCb !== undefined) {
+    authEventHandler.onErrorHandler = onErrorCb;
+  }
+  if (onServerError !== undefined) {
+    authEventHandler.onServerErrorHandler = onServerError;
+  }
+}

```

Comment:
This could be simplified to 
`authEventHandler.onServerErrorHandler = authEventHandler.onServerErrorHandler || onServerError;`

Reply:
even better to: 
`authEventHandler = {
    onSuccessfulLoginHandler || authEventHandler.onSuccessfulLoginHandler,
    onErrorHandler || authEventHandler.onErrorHandler,
    onServerErrorHandler || authEventHandler.onServerErrorHandler,
}`

Reply:
Whatever you like best not a big deal. Leave it be if you want :D

Reply:
I am going to take the first approach.